app/packs/controllers/date_range_controller.js

```diff
@@ -1,11 +1,107 @@
 import { Controller } from "stimulus"
 
 export default class extends Controller {
-  static targets = ["year"]
+    static targets = ["startYear", "startMonth", "startDay", "endYear", "endMonth", "endDay", "startError", "endError"]
 
-  // If any of the years are filled in, then make both required.
-  change(_event) {
-   const filledIn = this.yearTargets.find((target) => target.value !== '')
-   this.yearTargets.forEach((target) => target.required = filledIn)
-  }
+    connect() {
+        this.change()
+    }
+
+    // This invokes Date Validation controller as necessary for start and end dates.
+    change() {
+        this.clear()
+        this.validate()
+    }
+
+    validate() {
+        this.startYearTarget.dispatchEvent(new Event('validate'))
+        this.endYearTarget.dispatchEvent(new Event('validate'))
+        const filled = this.dateFilled()
+        if(filled) this.dateOrder()
+    }
+
+    clear() {
```

Comment:
Can this have a more descriptive name?  `clearValidationMessages()` maybe?

Reply:
Done.